packages/xml/src/decorators.ts

```diff
@@ -0,0 +1,118 @@
+import {
+  $encodedName,
+  DecoratorContext,
+  Enum,
+  ModelProperty,
+  Program,
+  Type,
+} from "@typespec/compiler";
+import {
+  AttributeDecorator,
+  NameDecorator,
+  NsDeclarationsDecorator,
+  NsDecorator,
+  UnwrappedDecorator,
+} from "../generated-defs/TypeSpec.Xml.js";
+import { XmlStateKeys, reportDiagnostic } from "./lib.js";
+import { XmlNamespace } from "./types.js";
+
+export const namespace = "TypeSpec.Xml";
+
+export const $name: NameDecorator = (context, target, name) => {
+  context.call($encodedName, target, "application/xml", name);
+};
+
+export const $attribute: AttributeDecorator = (context, target) => {
+  context.program.stateSet(XmlStateKeys.attribute).add(target);
+};
+
+export function isAttribute(program: Program, target: ModelProperty): boolean {
+  return program.stateSet(XmlStateKeys.attribute).has(target);
+}
+
+export const $unwrapped: UnwrappedDecorator = (context, target) => {
+  context.program.stateSet(XmlStateKeys.unwrapped).add(target);
+};
+
+export function isUnwrapped(program: Program, target: ModelProperty): boolean {
+  return program.stateSet(XmlStateKeys.unwrapped).has(target);
+}
+
+export const $nsDeclarations: NsDeclarationsDecorator = (context, target) => {
+  context.program.stateSet(XmlStateKeys.nsDeclaration).add(target);
+};
+
+function isNsDeclarationsEnum(program: Program, target: Enum): boolean {
```

Comment:
I feel like we should make it a habit to have jsdoc comments for exported types and accessor functions.  These are all pretty straightforward, but it would be nice to have. Only the ns accessors might *actually* need documentation, though

Reply:
Yeah we should do that, I’ll update them